pkg/controller/atlasdeployment/serverless_deployment.go

```diff
@@ -40,8 +46,26 @@ func ensureServerlessInstanceState(ctx *workflow.Context, project *mdbv1.AtlasPr
 
 	switch atlasDeployment.StateName {
 	case status.StateIDLE:
+		convertedDeployment, err := serverlessSpec.ServerlessToAtlas()
+		if err != nil {
+			return atlasDeployment, workflow.Terminate(workflow.Internal, err.Error())
+		}
+		if convertedDeployment.Tags == nil {
+			convertedDeployment.Tags = &[]*mongodbatlas.Tag{}
+		}
+		if !reflect.DeepEqual(convertedDeployment.Tags, atlasDeployment.Tags) {
```

Comment:
And turns out we do not want reflect here I think, as it makes this check incorrect:
- []TagSpec is an array, so order is important and DeepEqual will take it into account.
- But in reality, we do not care if x tags appear in the same or different order, the tag set is the same right?

I would make a custom function for the check and unit test it as per requirements.

Reply:
I'd second José here and discourage the usage of reflect in the code.

Reply:
@cveticm it looks like this has to be converted into a custom comparison function, as `reflect.DeepEqual()` takes into account the positions of array elements:
```
	t1 := []*TagSpec{&TagSpec{Key: "1", Value: "2"}, &TagSpec{Key: "2", Value: "3"}}
	t2 := []*TagSpec{&TagSpec{Key: "2", Value: "3"}, &TagSpec{Key: "1", Value: "2"}}
```
These two won't be equal for `DeepEqual`